### ISSUE TITLE
Improve follow flow with custom web protocols (web+follow:username@host.tld)

### ISSUE THREAD

**Issue by pfefferle** [tootsuite/mastodon#14187]
### Pitch

In 2010 Mozilla wrote an article "[RegisterProtocolHandler Enhancing the Federated Web](https://blog.mozilla.org/webdev/2010/07/26/registerprotocolhandler-enhancing-the-federated-web/)".

The idea is to register a protocol handler when logged into a mastodon account:

```JavaScript
if (typeof navigator.registerProtocolHandler === "function") {
    navigator.registerProtocolHandler("web+follow", "https://mastodon.social/follow?uri=%s", title);
}
```

On the profile pages, the follow button could look like:

```HTML
<!-- check if browser support registerProtocolHandler -->
<a href="web+follow:username@host.tld">Follow</a>
```

`RegisterProtocolHandler` is supported by [most modern browsers](https://caniuse.com/#feat=registerprotocolhandler) (except Safari but maybe Mastodon could change this). If there is a check, it should not break anything.

### Motivation

Improve the "decentralized follow" flow, to feel more like a classic silo (twitter, facebook) follow.

**Comment by ThibG** [comment]
We used to register protocol handlers but stopped doing that because of a combination of it not being actually used much, and browser's UI being off-putting: #8127

**Comment by pfefferle** [comment]
Oh, so sad. I think it is Hen/Egg problem. If there is no service using it, I think it will not improve.

**Comment by SuperSandro2000** [comment]
This should be re-added but not after 5 minutes but with a button at a good place.

**Comment by SoniEx2** [comment]
:+1: to having a button in a good place. I'd put it under preferences.

however, a bigger issue is that websites can't just provide a fallback handler like they can provide base URLs for anchors. see also https://github.com/whatwg/html/issues/4584

**Comment by pfefferle** [comment]
@SoniEx2 you are right... the implementation is far from perfect, but I think a platform like mastodon (or the complete fediverse) would (hopefully) give some importance to the custom protocol handler and would help to improve the spec and the current browser implementations.

...but I would recommend to not use `web+mastodon:` ...it would be awesome to have a generic "follow" button for the fediverse! 😍

**Comment by SoniEx2** [comment]
yeah. `web+activitypub:` would be nice, I think.

**Comment by DarkArc** [comment]
I've added a suggestion on the Mozilla side for first party `activitypub://foo@bar.com` URL protocol handler support: 

https://connect.mozilla.org/t5/ideas/add-user-friendly-first-party-activitypub-handle-support/idi-p/19622
